n3fit/src/n3fit/layers/DY.py

```diff
@@ -84,8 +84,9 @@ def compute_dy_observable_many_replica(pdf, padded_fk):
         tensor
             observable of shape (batch=1, replicas, ndata)
     """
-    temp = op.einsum('nxfyg, bryg -> brnxf', padded_fk, pdf)
-    return op.einsum('brnxf, brxf -> brn', temp, pdf)
+    # TODO: check which PDF must go first in case of different initial states!!!
+    temp = op.einsum('nxfyg, bryg -> brnxf', padded_fk, pdf[0])
+    return op.einsum('brnxf, brxf -> brn', temp, pdf[1])
```

Comment:
```suggestion
    pdfa = pdf[1]
    pdfb = pdf[0]
    
    temp = op.einsum('nxfyg, bryg -> brnxf', padded_fk, pdfa)
    return op.einsum('brnxf, brxf -> brn', temp, pdfb)
```

I still need to check but I think this is the right ordering.

Reply:
Good catch!!! Will push this.